asreview/webapp/api/auth.py

```diff
@@ -0,0 +1,512 @@
+
+    if not user:
+        # user does not exsist
+        result = (404, {"message": f"User account {email} does not exist."})
+    elif not user.confirmed:
+        # account is not confirmed
+        result = (404, {"message": f"User account {email} is not confirmed."})
+    else:
+        # user exists and is confirmed: verify password
+        if user.verify_password(password):
+            logged_in = perform_login_user(user)
+            result = (
+                200,
+                {"logged_in": logged_in, "name": user.get_name(), "id": user.id},
+            )
+        else:
+            # password is wrong
+            if user.origin == "asreview":
+                # if this is an asreview user
+                result = (404, {"message": f"Incorrect password for user {email}"})
+            else:
+                # this must be an OAuth user trying to get in with
+                # a password
+                service = user.origin.capitalize()
+                result = (404, {"message": f"Please login with the {service} service"})
+
+    status, message = result
+    response = jsonify(message)
+    return response, status
+
+
+@bp.route("/signup", methods=["POST"])
+def signup():
+    # this is for the response
+    user_id = False
+
+    # Can we create accounts?
+    if current_app.config.get("ALLOW_ACCOUNT_CREATION", False):
+        email = request.form.get("email", "").strip()
+        name = request.form.get("name", "").strip()
+        affiliation = request.form.get("affiliation", "").strip()
+        password = request.form.get("password")
+        public = bool(int(request.form.get("public", "1")))
+
+        # check if email already exists
+        user = User.query.filter(
+            or_(User.identifier == email, User.email == email)
+        ).one_or_none()
+        # return error if user doesn't exist
+        if isinstance(user, User):
+            result = (403, f'User with email "{email}" already exists.')
+        else:
+            # password confirmation is done by front end, so the only
+            # thing that remains is to add the user (password will be
+            # hashed in the User model)
+            try:
+                identifier = email
+                origin = "asreview"
+                # are we going to verify the email?
+                email_verification = bool(
+                    current_app.config.get("EMAIL_VERIFICATION", False)
+                )
+                # set confirmed to False if we 'do' verification. Note
+                # that this route only creates 'asreview' accounts
+                confirmed = not email_verification
+                # create the User account
+                user = User(
+                    identifier=identifier,
+                    origin=origin,
+                    email=email,
+                    name=name,
+                    affiliation=affiliation,
+                    password=password,
+                    confirmed=confirmed,
```

Comment:
It's probably more clear to use `not email_verification` remove the `confirmed` variable. I don't like the double negation.